File: javascript/7_kyu/fix_string_case.js
// Fix string case
// https://www.codewars.com/kata/5b180e9fedaa564a7000009a/train/javascript

function solve(s){
    var uppercase = 0;
    var lowercase = 0;
    
    for (var i = 0; i < s.length; i++) {
      if (s[i] == s[i].toUpperCase()) {
        uppercase++;
      } else {
        lowercase++;
      }
    }
    
    var ans = [];
    
    if (lowercase >= uppercase) {
      for (var i = 0; i < s.length; i++) {
        if (s[i] == s[i].toUpperCase()) {
          ans.push(s[i].toLowerCase());
        } else {
          ans.push(s[i]);
        }
      }
    } else if (lowercase < uppercase) {
      for (var i = 0; i < s.length; i++) {
        if (s[i] == s[i].toUpperCase()) {
          ans.push(s[i]);
        } else {
          ans.push(s[i].toUpperCase());
        }
      }
    }
    
    return ans.join('');
  }

File: python/7_kyu/fix_string_case.py
# Fix string case
# https://www.codewars.com/kata/5b180e9fedaa564a7000009a/train/python

def solve(s):
    uppercase = 0
    lowercase = 0
    
    for i in s:
        if i.isupper():
            uppercase += 1
        else:
            lowercase += 1
    
    ans = []
    
    if lowercase >= uppercase:
        for i in s:
            if i.isupper():
                ans.append(i.lower())
            else:
                ans.append(i)
        return ''.join(ans)
    elif lowercase < uppercase:
        for i in s:
            if i.islower():
                ans.append(i.upper())
            else:
                ans.append(i)
        return ''.join(ans)